### ISSUE TITLE
PioniaCors: Missing Symfony\Component\HttpFoundation\Response Object

### ISSUE THREAD

**Issue by geferson0000** [JetPhp-project/JetPhp-Framework#3]
### Which version php are you using
8.3.8
### Pionia version
1.0.0
### Node version
22.11.0
### Vite version
4.3.9

This is an example app I created to reproduce the error

``` php
<?php

  

namespace Application\Services;

  

use Exception;

use Pionia\Collections\Arrayable;

use Pionia\Http\Response\BaseResponse;

use Pionia\Http\Services\Service;

# TodoService.php

class TodoService extends Service

{
    protected function listAction(Arrayable $data): BaseResponse

    {

        return response(0, null, table('todos')->all());

    }
}

```

``` javascript
import { defineConfig } from 'vite'

import AlpineVitePlugin from './alpine-vite-plugin.js'

// vite.config.js

export default defineConfig({

    server: {

        proxy: {

            '/api': {

                target: 'http://localhost:8000',

                changeOrigin: true,

                rewrite: (path) => path.replace(/^\/api/, '')

            }

        }

    },

    plugins: [],

})
```

``` javascript

// data.js

const options = {

    method: 'POST',

    headers: {'Content-Type': 'application/x-www-form-urlencoded'},

    body: new URLSearchParams({service: 'todo', action: 'list', id: '1'})

  };

  fetch('api/v1/', options)

    .then(response => response.json())

    .then(response => console.log(response))

    .catch(err => console.log(err));
```

``` html
<!DOCTYPE html>

<html lang="en">

<head>

  <meta charset="UTF-8">

  <meta name="viewport" content="width=device-width, initial-scale=1.0">

  <script src="./data.js"></script>

  <title>Document</title>

</head>

<body>

  <!-- index.html -->

</body>

</html>
```

I having this error in pionia console 

``` console

[Wed Dec 11 16:24:00 2024] [2024-12-11 20:24:00] pionia.ERROR >> Error handling request POST::http://localhost:8000/v1/  {
    "error": "Fruitcake\\Cors\\CorsService::varyHeader(): Argument #1 ($response) must be of type Symfony\\Component\\HttpFoundation\\Response, null given, called in C:\\Users\\Dystech\\Documents\\repositories\\pionia\\pionia\\vendor\\pionia\\pionia-core\\src\\Pionia\\Cors\\PioniaCors.php on line 66"
}

```

in the browser
``` javascript
{
    "returnCode": 500,
    "returnMessage": "Fruitcake\\Cors\\CorsService::varyHeader(): Argument #1 ($response) must be of type Symfony\\Component\\HttpFoundation\\Response, null given, called in C:\\Users\\Dystech\\Documents\\repositories\\pionia\\pionia\\vendor\\pionia\\pionia-core\\src\\Pionia\\Cors\\PioniaCors.php on line 66",
    "returnData": null,
    "extraData": null
}
```

**Comment by jet2018** [comment]
Are you sure you're running on Pionia v1.0.0?

I don't think so. Please update to add the exact version you're running on. Thanks

**Comment by geferson0000** [comment]
I have run composer update in the terminal and the composer.json has been updated
composer.json updated
``` json
{
    "name": "pionia/pionia-app",
    "description": "Making Restful API development more restful \uD83D\uDCA4",
    "type": "project",
    "license": "MIT",
    "autoload": {
        "psr-4": {
            "Application\\": "",
            "Application\\Services\\": "services/",
            "Application\\Switches\\": "switches/",
            "Application\\Authentications\\": "authentications/",
            "Application\\Middlewares\\": "middlewares/",
            "Application\\Commands\\": "commands/"
        }
    },
    "authors": [
        {
            "name": "jet2018",
            "email": "ezrajet9@gmail.com"
        }
    ],
    "minimum-stability": "stable",
    "prefer-stable": true,
    "require": {
        "php": ">=8.1",
        "ext-pdo": "*",
        "pionia/pionia-core": "^2.0"
    },
    "scripts": {
        "post-create-project-cmd": [
            "php storage/scripts/rename.php"
        ]
    }
}

````

However, it presents the same problem

**Comment by geferson0000** [comment]
here is the pionia code https://github.com/geferson0000/pionia.git

**Comment by geferson0000** [comment]
but it works correctly when the request is direct
![image](https://github.com/user-attachments/assets/7c83d727-ae80-46e4-8462-f8bf9d4a2dc6)

**Comment by jet2018** [comment]
Can we maintain this discussion on the issue itself coz there is someone working on this issue, he might need this information

Tumusiime Ezra

SERVICE COPS LTD
________________________________

**Comment by geferson0000** [comment]
> Can we maintain this discussion on the issue itself coz there is someone working on this issue, he might need this information

Ok... I'll answer you directly from the 'issue'

**Comment by jet2018** [comment]
Thanks for your contributions so far. I hope you can help us resolve those small issues and get this framework to work out there

Tumusiime Ezra

SERVICE COPS LTD
________________________________

**Comment by hakrichTech** [comment]
please check the cros configuration

**Comment by jet2018** [comment]
We were able to reproduce this issue, that's an awesome step bytheway, and we shall be resolving it soon

**Comment by jet2018** [comment]
Version 2.0.4 fixes #3, please upgrade by running `composer upgrade` to resolve this issue.
